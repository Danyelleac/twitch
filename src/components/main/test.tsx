import { render, screen } from '@testing-library/react'

import { Main } from '.'

describe('<Main/>', () => {
  it('should render the heading', () => {
    //renderizar o componente (render)
    render(<Main />)
    //buscar o elemento (screen) - e verificar se ele existe (getByRole)
    expect(
      screen.getByRole('heading', { name: /react avançado/i })
    ).toBeInTheDocument()
  })
  it('should render snapshot', () => {
    //renderizar o componente (render)
    const { container } = render(<Main />)
    //gerar um arquivo de snapshot
    expect(container.firstChild).toMatchSnapshot() //verifica se o primeiro elemento é igual ao snapshot
  })

  it('should render the colors correctly', () => {
    //renderizar o componente (render)
    const { container } = render(<Main />)
    //verificar se o background-color está correto
    expect(container.firstChild).toHaveStyle({
      //verifica se o primeiro elemento tem o estilo igual ao snapshot
      'background-color': '#06092b'
    })
  })
})
