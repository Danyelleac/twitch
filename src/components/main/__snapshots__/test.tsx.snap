// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
  height: 100%;
  width: 100%;
  background-color: #06092b;
  color: #fff;
  padding: 3rem;
}

<main
  class="c0"
>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
